mail_notification_with_history/models/mail_message.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class Message(models.Model):
+    _inherit = "mail.message"
+
+    def get_notification_message_history(self):
+        """Get the list of messages to include into an email notification history."""
+        if not self.env[self.model].with_mail_notification_history():
+            return []
+        if not self._with_notification_message_history():
+            return []
+        domain = self._get_notification_message_history_domain()
+        messages = self.env["mail.message"].search(domain, order="date desc")
+        return messages - self
+
+    def _get_notification_message_history_domain(self):
+        return [
+            ("model", "=", self.model),
+            ("res_id", "=", self.res_id),
+            ("message_type", "in", ("user_notification", "comment", "email")),
+        ]
+
+    def _with_notification_message_history(self):
```

Comment:
```suggestion
    def _should_include_notification_message_history(self):
```

Just a suggestion. Feel free to ignore it.

IMO `with_*` implies some kind of change in the recordset. Makes me think of `with_company`, `with_user`, `with_prefetch`...

Reply:
Well I removed this one as it is not really needed.